services/director-v2/src/simcore_service_director_v2/modules/project_networks.py

```diff
@@ -0,0 +1,289 @@
+) -> bool:
+    decoded_service_key = urllib.parse.unquote_plus(service_key)
+
+    # check the type of service, if not dynamic do
+    # not fetch the labels
+    # (simcore)/(services)/(comp|dynamic|frontend)
+    service_type = decoded_service_key.split("/")[2]
+    if service_type != "dynamic":
+        return False
+
+    simcore_service_labels: SimcoreServiceLabels = (
+        await director_v0_client.get_service_labels(
+            service=ServiceKeyVersion(key=decoded_service_key, version=service_version)
+        )
+    )
+    return simcore_service_labels.needs_dynamic_sidecar
+
+
+async def _send_network_configuration_to_dynamic_sidecar(
+    scheduler: DynamicSidecarsScheduler,
+    project_id: ProjectID,
+    new_networks_with_aliases: NetworksWithAliases,
+    existing_networks_with_aliases: NetworksWithAliases,
+) -> None:
+    """
+    Propagates the network configuration to the dynamic-sidecars.
+    Note:
+    - All unused networks will be removed from each service.
+    - All required networks will be attached to each service.
+    - Networks which are in use will not be removed.
+    """
+
+    # REMOVING
+    to_remove_items: Set[_ToRemove] = set()
+
+    # if network no longer exist remove it from all nodes
+    for new_network_name, node_ids_and_aliases in new_networks_with_aliases.items():
+        if new_network_name not in existing_networks_with_aliases:
+            for node_id in node_ids_and_aliases:
+                to_remove_items.add(
+                    _ToRemove(
+                        project_id=project_id,
+                        node_id=node_id,
+                        network_name=new_network_name,
+                    )
+                )
+    # if node does not exist for the network, remove it
+    # if alias is different remove the network
+    for new_network_name, node_ids_and_aliases in new_networks_with_aliases.items():
+        existing_node_ids_and_aliases = existing_networks_with_aliases.get(
+            new_network_name, {}
+        )
+        for node_id, alias in node_ids_and_aliases.items():
+            # node does not exist
+            if node_id not in existing_node_ids_and_aliases:
+                to_remove_items.add(
+                    _ToRemove(
+                        project_id=project_id,
+                        node_id=node_id,
+                        network_name=new_network_name,
+                    )
+                )
+            else:
+                existing_alias = existing_networks_with_aliases[new_network_name][
+                    node_id
+                ]
+                # alias is different
+                if existing_alias != alias:
+                    to_remove_items.add(
+                        _ToRemove(
+                            project_id=project_id,
+                            node_id=node_id,
+                            network_name=new_network_name,
+                        )
+                    )
+
+    await logged_gather(
```

Comment:
Does this really need a "fail-fast" policy here? I mean, one failure stops entire operation. Would not be more suitable a "try-all-and-fail" instead?

Reply:
I remember why this is/was like this, before being moved form `webserver` to `director-v2`. "Connectivity" errors are already handled downstream. This is OK like this.